pkg/authorization/util/util.go

```diff
@@ -0,0 +1,143 @@
+package util
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/rbac"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/authorization/api"
+)
+
+func ConvertOriginClusterRole(in *api.ClusterRole) *rbac.ClusterRole {
+	return &rbac.ClusterRole{
+		ObjectMeta: in.ObjectMeta,
+		Rules:      convertOriginPolicyRule(in.Rules),
+	}
+}
+
+func ConvertOriginRole(in *api.Role) *rbac.Role {
+	return &rbac.Role{
+		ObjectMeta: in.ObjectMeta,
+		Rules:      convertOriginPolicyRule(in.Rules),
+	}
+}
+
+func ConvertOriginClusterRoleBinding(in *api.ClusterRoleBinding) *rbac.ClusterRoleBinding {
+	return &rbac.ClusterRoleBinding{
+		ObjectMeta: in.ObjectMeta,
+		Subjects:   convertOriginSubjects(in.Subjects),
+		RoleRef:    convertOriginRoleRef(&in.RoleRef),
+	}
+}
+
+func ConvertOriginRoleBinding(in *api.RoleBinding) *rbac.RoleBinding {
+	return &rbac.RoleBinding{
+		ObjectMeta: in.ObjectMeta,
+		Subjects:   convertOriginSubjects(in.Subjects),
+		RoleRef:    convertOriginRoleRef(&in.RoleRef),
+	}
+}
+
+func convertOriginPolicyRule(in []api.PolicyRule) []rbac.PolicyRule {
+	rules := make([]rbac.PolicyRule, 0, len(in))
+	for _, rule := range in {
+		r := rbac.PolicyRule{
+			APIGroups:       rule.APIGroups,
+			Verbs:           rule.Verbs.List(),
+			Resources:       rule.Resources.List(),
+			ResourceNames:   rule.ResourceNames.List(),
+			NonResourceURLs: rule.NonResourceURLs.List(),
+		}
+		rules = append(rules, r)
+	}
+	return rules
+}
+
+func convertOriginSubjects(in []kapi.ObjectReference) []rbac.Subject {
+	subjects := make([]rbac.Subject, 0, len(in))
+	for _, subject := range in {
+		s := rbac.Subject{
+			Kind:       subject.Kind,
```

Comment:
origin kinds collapse down to just `User` and `Group` from `SystemUser` and `SystemGroup`.